File: day_04/day4a.js
const fs = require('fs');

const inputText = fs.readFileSync('./day_04/day4_input.txt', 'utf-8');
const rows = inputText.split('\n');

var sum = 0
rows.forEach((row, index) => {
    cutCard = row.slice(8);
    const parts = cutCard.split('|');

    const convertToArray = (str) => {
        return str.trim().split(/\s+/).map(num => parseInt(num, 10));
    };
    const winning = convertToArray(parts[0]);
    const ours = convertToArray(parts[1]);
    var num_wins = 0
    ours.forEach(num => {
        if (winning.includes(num)) {
            num_wins += 1;
        };
    });
    sum += Math.floor(2**(num_wins-1))
})
console.log(sum)

File: day_04/day4a.py
import math

with open("day_04/day4_input.txt", 'r') as file:
    characters_array = [line.strip() for line in file]

sum = 0
for row in characters_array:
    split_array = row.split('|')
    winning = [int(num) for num in split_array[0].split() if num.isdigit()]
    ours = [int(num) for num in split_array[1].split() if num.isdigit()]
    num_wins = 0
    for num in ours:
        if num in winning:
            num_wins += 1
    sum += math.floor(2**(num_wins-1))
print(sum)

